sql/search_derived/search_metric_contribution/query.sql

```diff
@@ -0,0 +1,213 @@
+-- temp function returns how much of one metric is contributed by the quantile of another metric
+CREATE TEMP FUNCTION get_quantile(metric1 FLOAT64, metric2 FLOAT64, quantile FLOAT64) AS (
+  IF(metric1 > quantile, metric2, NULL)
+);
+
+WITH monthly_client_aggregates AS (
+  SELECT
+    client_id,
+    SUM(sap) AS sap,
+    SUM(search_with_ads) AS search_with_ads,
+    SUM(active_hours_sum) AS active_hours_sum,
+    SUM(ad_click) AS ad_click,
+  FROM
+    `moz-fx-data-shared-prod.search.search_clients_daily`
+  WHERE
+    submission_date >= DATE_SUB(@submission_date, INTERVAL 27 DAY)
+  GROUP BY
+    1
+),
+monthly_aggregates AS (
```

Comment:
This and the CTE below can be merged into one (with all `SUM` fields and all `APPROX_QUANTILE` fields).